builtin/logical/pki/path_revoke.go

```diff
@@ -461,26 +479,23 @@ func validatePrivateKeyMatchesCert(signer crypto.Signer, certReference *x509.Cer
 	// See: https://github.com/golang/go/blob/c6a2dada0df8c2d75cf3ae599d7caed77d416fa2/src/crypto/tls/tls.go#L304-L331
 	switch certPub := certReference.PublicKey.(type) {
 	case *rsa.PublicKey:
-		privPub, ok := signer.Public().(*rsa.PublicKey)
+		privPub, ok := verifier.(*rsa.PublicKey)
 		if !ok {
 			return errutil.UserError{Err: "provided private key type does not match certificate's public key type"}
 		}
-		if err := signer.(*rsa.PrivateKey).Validate(); err != nil {
-			return err
-		}
 		if certPub.N.Cmp(privPub.N) != 0 || certPub.E != privPub.E {
 			return errutil.UserError{Err: "provided private key does not match certificate's public key"}
```

Comment:
and again below here - privateKey shouldn't be referenced in publicKey Function?